### DISCUSSION TITLE
How to check whether a workspace is enabled?

### DISCUSSION THREAD

**Discussion by medwatt** [i3/i3#4805] [Q&A] [answered]
I would like to be able to know whether a workspace is open. If it is, then focus the workspace. If not, then focus the workspace and open an app. How can I do that?

**Comment by Airblader** [comment, marked as answer]
You can use the `GET_WORKSPACES` IPC request to get the list of current workspaces (eg `i3-msg -t get_workspaces`). Process that JSON to look for a workspace of the specific name (eg using `jq`).

**Reply by medwatt** [reply]
Thanks for the answer. I have come up with this solution for now. As I have just recently started using i3, I don't know if it is a good solution. Is there a way to improve it?

```bash
#!/bin/bash

if [ $(i3-msg -t get_workspaces | grep $1 | wc -l) = 0 ]; then
    i3-msg "workspace $1; exec $2";
else
    i3-msg "workspace $1";
fi
```

Then:

```
bindsym $super+n exec "~/.config/i3/scripts/open-workspace.sh nvim 'kitty nvim'"
```

**Reply by Airblader** [reply]
Grepping in structured output like JSON is very error prone. That's why I suggested using `jq`.

**Reply by medwatt** [reply]
Thanks once more. Do you know if something like this is possible?

```bash
set $script "~/.config/i3/scripts/open-workspace.sh"

bindsym $super+n exec $script nvim 'kitty nvim'
```

**Reply by Airblader** [reply]
It is, but you're probably going to want to remove or change the double quotes. Variables in i3 are basically substituted verbatim, in the most stupendously simple way you can imagine. With your snippet, what i3 will actually see is

```
bindsym $super+n exec "~/.config/i3/scripts/open-workspace.sh" nvim 'kitty nvim'
```

… which has the quotes placed incorrectly.
